src/gui/plugins/apply_force_torque/ApplyForceTorque.cc

```diff
@@ -0,0 +1,321 @@
+        if (link.Entity() == this->dataPtr->selectedEntity)
+        {
+          this->dataPtr->linkIndex = i;
+        }
+        ++i;
+      }
+    }
+  }
+
+  if (this->dataPtr->changedIndex)
+  {
+    this->dataPtr->changedIndex = false;
+
+    if (this->dataPtr->selectedEntity.has_value())
+    {
+      auto parentModel = Link(*this->dataPtr->selectedEntity).ParentModel(_ecm);
+      std::string linkName =
+        this->dataPtr->linkNameList[this->dataPtr->linkIndex].toStdString();
+      this->dataPtr->selectedEntity = parentModel->LinkByName(_ecm, linkName);
+    }
+  }
+
+  // Get the position of the center of mass
+  if (this->dataPtr->selectedEntity.has_value())
+  {
+    auto linkWorldPose = worldPose(*this->dataPtr->selectedEntity, _ecm);
+    auto inertial = _ecm.Component<components::Inertial>(
+      *this->dataPtr->selectedEntity);
+    if (inertial)
+    {
+      this->dataPtr->inertialPos =
+        linkWorldPose.Rot().RotateVector(inertial->Data().Pose().Pos());
+    }
+  }
+
+  emit this->ModelNameChanged();
+  emit this->LinkNameListChanged();
+  emit this->LinkIndexChanged();
+}
+
+/////////////////////////////////////////////////
+QString ApplyForceTorque::ModelName() const
+{
+  return this->dataPtr->modelName;
+}
+
+/////////////////////////////////////////////////
+QStringList ApplyForceTorque::LinkNameList() const
+{
+  return this->dataPtr->linkNameList;
+}
+
+/////////////////////////////////////////////////
+int ApplyForceTorque::LinkIndex() const
+{
+  return this->dataPtr->linkIndex;
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::SetLinkIndex(int _linkIndex)
+{
+  this->dataPtr->linkIndex = _linkIndex;
+  this->dataPtr->changedIndex = true;
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::UpdateForce(double _x, double _y, double _z)
+{
+  this->dataPtr->force = {_x, _y, _z};
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::UpdateTorque(double _x, double _y, double _z)
+{
+  this->dataPtr->torque = {_x, _y, _z};
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::ApplyForce()
+{
+  this->dataPtr->PublishWrench(true, false);
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::ApplyTorque()
+{
+  this->dataPtr->PublishWrench(false, true);
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorque::ApplyAll()
+{
+  this->dataPtr->PublishWrench(true, true);
+}
+
+/////////////////////////////////////////////////
+void ApplyForceTorquePrivate::PublishWrench(bool _applyForce, bool _applyTorque)
+{
+  std::lock_guard<std::mutex> lock(this->mutex);
+
+  if (!this->selectedEntity.has_value())
+  {
+    gzdbg << "No link selected" << std::endl;
+    return;
+  }
+
+  math::Vector3d forceToApply =
+    _applyForce ? this->force : math::Vector3d::Zero;
+  math::Vector3d torqueToApply =
+    _applyTorque ? this->torque : math::Vector3d::Zero +
```

Comment:
Parenthesize the ternary expression. Otherwise, I'm not sure if the cross product term is being applied in both conditions.